rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -121,6 +124,21 @@ public void deleteAllProjects(User sw360User) {
         }
     }
 
+    public Set<String> getReleaseIds(String projectId, User sw360User, String transitive) {
+        try {
+            ProjectService.Iface sw360ProjectClient = getThriftProjectClient();
+            if (Boolean.parseBoolean(transitive)) {
+                List<ReleaseClearingStatusData> releaseClearingStatusData = sw360ProjectClient.getReleaseClearingStatuses(projectId, sw360User);
+                return releaseClearingStatusData.stream().map(r -> r.release.getId()).collect(Collectors.toSet());
+            } else {
+                final Project project = getProjectForUserById(projectId, sw360User);
+                return project.getReleaseIdToUsage().keySet();
+            }
+        } catch (TException e) {
+            throw new RuntimeException(e);
```

Comment:
I'm not a fan of throwing a generic exception. I think `getReleaseIds()` should declare `throws TException` and `getProjectReleases()` should catch the exception and return a proper `ResponseEntity` with HTTP 500.

Reply:
fixed exception handling for controller and service classes